protelis/protelis-interpreter/src/main/java/org/protelis/lang/datatype/impl/DeviceUUID.java

```diff
@@ -0,0 +1,60 @@
+package org.protelis.lang.datatype.impl;
+
+import java.util.UUID;
+
+import org.protelis.lang.datatype.DeviceUID;
+
+/** DeviceUID based on UUIDs. */
+public class DeviceUUID implements DeviceUID, Comparable<DeviceUUID> {
+    private static final long serialVersionUID = 1L;
+    private final UUID uid;
+
+    /**
+     * Generate a random UUID.
+     */
+    public DeviceUUID() {
+        this(UUID.randomUUID());
+    }
+
+    /**
+     * @param uid
+     *            the string to use as the UID
+     */
+    public DeviceUUID(final UUID uid) {
+        this.uid = uid;
+    }
+
+    /**
+     * 
+     * @return the underlying string
+     */
+    public UUID getUID() {
+        return uid;
+    }
+
+    @Override
+    public boolean equals(final Object alt) {
+        if (this == alt) {
+            return true;
+        } else if (alt instanceof DeviceUUID) {
```

Comment:
Sure

imagine having `MyDeviceUID extends StringUID`, which does not override `equals`. Now, you could write the following (using Xtend notation for brevity):
```xtend
val suid = new StringUID("foo")
val myuid = new MyDeviceUID("foo")
suid.equals(myuid) // true
myuid.equals(suid) // false
```
this is due to myuid being instance of StringUID and suid not being instance of MyUID. It violates the equals contract.

This could be fixed by comparing the object classes and returning false if they are not the same.